index/scorch/segment_plugin.go

```diff
@@ -26,25 +28,63 @@ import (
 	zapv15 "github.com/blevesearch/zapx/v15"
 )
 
-var supportedSegmentPlugins map[string]map[uint32]segment.Plugin
-var defaultSegmentPlugin segment.Plugin
+// Plugin represents the essential functions required by a package to plug in
+// it's segment implementation
+type Plugin interface {
```

Comment:
Feels like this name `Plugin` becomes a very generic name now, without leaving any clues?
Any thoughts?

Reply:
Originally I was going make it not exported, but we use it as an argument to the `RegisterPlugin` method.  We could change it to `SegmentPlugin`.  But, then should also change the method names to match?

Reply:
Alternatively, if you don't like `Plugin` at all, maybe `SegmentImplementation`?  That ends up kind of long, and `SegmentImpl` feels very much like a Java name...

Reply:
I liked `SegmentPlugin` though. .

Reply:
OK, I will go with `SegmentPlugin`, do you think the method names should change as well?  

RegisterPlugin -> RegisterSegmentPlugin
ResetPlugins -> ResetSegmentPlugins